### PR TITLE
EN-6798/tx-integrity-checks

### PR DESCRIPTION
**Pull request by sasurobert**
verification of txValue and length of usernames

### CODE REVIEW

File: process/transaction/interceptedTransaction.go [hunk 1]
```diff
@@ -171,6 +171,12 @@ func (inTx *InterceptedTransaction) integrity() error {
 	if inTx.tx.Value.Sign() < 0 {
 		return process.ErrNegativeValue
 	}
+	if len(inTx.tx.RcvUserName) > 0 && len(inTx.tx.RcvUserName) != inTx.hasher.Size() {
+		return process.ErrInvalidUserNameLength
```
Comment (iulianpascalau) [inline process/transaction/interceptedTransaction.go:175]: could have wrapped the errors to know which field was bad

File: process/transaction/interceptedTransaction.go [hunk 2]
```diff
@@ -171,6 +171,12 @@ func (inTx *InterceptedTransaction) integrity() error {
...
+	}
+	if len(inTx.tx.SndUserName) > 0 && len(inTx.tx.SndUserName) != inTx.hasher.Size() {
```
Comment (bogdan-rosianu) [inline process/transaction/interceptedTransaction.go:177]: why hasher size? can't a username be longer or shorter?

Comment (sasurobert) [reply process/transaction/interceptedTransaction.go:177]: username is saved as a hash - thus 32 bytes always

File: process/economics/economicsData.go [hunk 3]
```diff
@@ -272,6 +272,14 @@ func (ed *EconomicsData) CheckValidityTxValues(tx process.TransactionWithFeeHand
 		return process.ErrHigherGasLimitRequiredInTx
 	}
 
+	if len(tx.GetValue().Bytes()) > len(ed.genesisTotalSupply.Bytes()) {
+		return process.ErrTxValueOutOfBounds
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/economics/economicsData.go:276]: in order to speed up processing. for super big ints.

**Pull request merged**